### PR TITLE
Add system vault navigation

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

Follow up of https://github.com/dust-tt/dust/pull/6745.

This PR adds the navigation for the system vault based on design from [here](https://www.figma.com/design/QOxHwppG64GtPocpDhS6Y7/Product?node-id=8382-44390&m=dev).

image

### CODE REVIEW

File: front/components/vaults/VaultSideBarMenu.tsx [hunk 1]
```diff
@@ -110,9 +110,102 @@ const SubItemIconItemWrapper = (
 
 // System vault.
 
-const SystemVaultMenu = () => {
-  // TODO(Groups UI) Implement system vault menu.
-  return <></>;
+const SYSTEM_VAULTS_ITEMS = [
```
Comment (JulesBelveze) [inline front/components/vaults/VaultSideBarMenu.tsx:113]: Should we move this to types? Feel like we are gonna need this in other places

File: front/components/vaults/VaultSideBarMenu.tsx [hunk 2]
```diff
@@ -205,13 +298,15 @@ const DATA_SOURCE_OR_VIEW_SUB_ITEMS: {
...
   owner,
   vault,
-  item,
+  viewType,
 }: {
+  item: DataSourceOrViewInfo;
   owner: LightWorkspaceType;
   vault: VaultType;
-  item: DataSourceOrViewInfo;
+  viewType: "data_sources" | "data_source_views";
```
Comment (JulesBelveze) [inline front/components/vaults/VaultSideBarMenu.tsx:309]: Should we move the `DataSourceOrView` type from swr to types and reuse it here?

**Review by JulesBelveze — APPROVED** [review]

File: front/components/vaults/VaultSideBarMenu.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/vaults/VaultSideBarMenu.tsx:113]: This is specifically for menu rendering so I'd keep it close to the implementation. What other places do you have in mind?

File: front/components/vaults/VaultSideBarMenu.tsx (hunk 2 above)
Comment (flvndvd) [reply front/components/vaults/VaultSideBarMenu.tsx:309]: Indeed this one could live in `types`. Let me move it.

File: front/components/vaults/VaultSideBarMenu.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/vaults/VaultSideBarMenu.tsx:114]: The company data screen?

**Pull request merged**